docs/scripts/buildApi.ts

```diff
@@ -817,18 +815,7 @@ async function buildDocs(options: {
     reactApi.styles.name = generateMuiName(reactApi.name);
   }
   reactApi.styles.classes.forEach((key) => {
-    reactApi.styles.globalClasses[key] = generateClassName(
-      // @ts-expect-error
-      {
-        key,
-      },
-      {
-        options: {
-          name: reactApi.styles.name || generateMuiName(name),
-          theme: {},
-        },
-      },
-    );
+    reactApi.styles.globalClasses[key] = generateUtilityClass(generateMuiName(name), key);
```

Comment:
Could we handle the docs in a different effort? It can be solved as a standalone issue. I also have a few doubt with how the name is used, there might be room for simplification.

Reply:
@oliviertassinari Done.

Reply:
Great, don't hesitate to spin this off as a standalone PR. The solution is not obvious and fixing this will help. Otherwise, we can open a new issue for it once #26830 is closed.